torch/csrc/autograd/python_function.cpp

```diff
@@ -89,7 +68,7 @@ auto PyNode::apply(variable_list&& inputs) -> variable_list {
   auto num_inputs = inputs.size();
   THPObjectPtr pyInputs(PyTuple_New(static_cast<Py_ssize_t>(num_inputs)));
   if (!pyInputs)
-    throw_python_error();
+    throw python_error();
```

Comment:
The idea here is that even when these functions are used during the backward, the engine is the one responsible for persisting the error. Never the user code?

Reply:
Yeah, but apparently not in all cases (in particular the distributed case). I'm reverting this change, and adding a todo in the comments to do as a follow up.